### PR TITLE
Add minor things

### LINKED ISSUE #190: Add weights parameter to add_value function in Histogram
I had the idea that it might be useful to have an optional weight parameter associated to the `add_value` function in the Histogram class.
Depending on the input, one would either have a single value as a weight or, if the input is a list of values, a list of weights.
Currently, this is not possible or only if one generates a histogram for each value and then performs a weighted average over the histograms. I think it's a useful thing to add, and it is also one functionality in the NumPy histogram.

@NGoetz, @nilssass, What do you think?

### PR DESCRIPTION
**Pull request by Hendrik1704**
- Adds missing CHANGELOG entries
- Adds weights to Histogram and also the possibility to create probability densities (closes #190)
- Improvement of documentation
- Throw error in JetAnalysis if status not set

### CODE REVIEW

File: src/sparkx/Histogram.py [hunk 1]
```diff
@@ -268,12 +271,25 @@ def add_value(self, value):
...
 
         Raises
         ------
         TypeError
             if the input is not a number or numpy.ndarray or list
+        ValueError
+            if the input `weight` has not the same dimension as `value`
         """
+        # Check if weight has the same length as value
+        if weight is not None:
```
Comment (NGoetz) [inline src/sparkx/Histogram.py:286]: Should we use `None` or `nan`?  What happens if I use particle properties as weights?

File: src/sparkx/Histogram.py [hunk 2]
```diff
@@ -300,20 +320,67 @@ def add_value(self, value):
...
+    def make_density(self):
+        """
+        Make a probability density from the last histogram.
+        The result represents the probability density function in a given bin. 
+        It is normalized such that the integral over the whole histogram 
+        yields 1. This behavior is similar to the one of numpy histograms.
+
+        Notes
+        -----
+        If this function is used, then the routines to compute uncertainties
```
Comment (NGoetz) [inline src/sparkx/Histogram.py:354]: Could we not continue using it if you were to use the `scale_histogram` method?

**Review by NGoetz — CHANGES_REQUESTED** [review]
I have two points one should discuss before merging.

File: src/sparkx/Histogram.py (hunk 1 above)
Comment (Hendrik1704) [reply src/sparkx/Histogram.py:286]: We will use None and build in a nan check for the weight value.

**Pull request merged**